lib/ansible/plugins/action/unarchive.py

```diff
@@ -38,9 +38,20 @@ def run(self, tmp=None, task_vars=None):
 
         source  = self._task.args.get('src', None)
         dest    = self._task.args.get('dest', None)
-        copy    = boolean(self._task.args.get('copy', True))
+        remote_src = boolean(self._task.args.get('remote_src', False))
         creates = self._task.args.get('creates', None)
 
+        # "copy" is deprecated in favor of "remote_src".
+        if 'copy' in self._task.args:
+            # They are mutually exclusive.
+            if 'remote_src' in self._task.args:
+                result['failed'] = True
+                result['msg'] = "parameters are mutually exclusive: ('copy', 'remote_src')"
+                return result
+            # We will take the information from copy and store it in
+            # the remote_src var to use later in the playbook
```

Comment:
> BTW That does not change the fact that the logic in unarchive still has to check both copy and remote_src in various places, which is overly complex for a deprecated option IMO. That's why I favored your implementation.

Yeah - the logic is implemented twice. Once in actions/unarchive.py and once in the unarchive.py in the module. I've been told modules will be merged into the main repo in the future so this code can all live in the same repo and perhaps share the logic for parsing the task's arguments. That would be a better solution.